### PR TITLE
fix(Interactive Video): Added Bumper to Interactive Video ads and connected events

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
[AAG-2901]

## DESCRIPTION
Added Bumper to Interactive Video ads and connected events, I have also duplicated `CustomWebView` and `ViewabilityDetector` from the Common module.

Please note, these changes only affect the old base part of the codebase as the common part appears to support this functionality.

## SCREENSHOTS

https://user-images.githubusercontent.com/618350/226976748-a3fcff1b-9bb7-428d-8408-25309b8cd485.mov

[AAG-2901]: https://superawesomeltd.atlassian.net/browse/AAG-2901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/tv/superawesome/demoapp/MyApplication.kt [hunk 1]
```diff
@@ -12,7 +13,7 @@ class MyApplication : MultiDexApplication() {
     override fun onCreate() {
         super.onCreate()
         AwesomeAds.init(this, true)
-        AwesomeAdsSdk.init(this, Configuration(logging = true))
+        AwesomeAdsSdk.init(this, Configuration(logging = true, environment = Environment.Staging))
```
Comment (gunhansancar) [inline app/src/main/java/tv/superawesome/demoapp/MyApplication.kt:16]: I think our test placements use production environment. Why did we change this?

File: superawesome-base/src/main/java/tv/superawesome/sdk/publisher/SAVideoActivity.java [hunk 2]
```diff
@@ -333,16 +333,16 @@ class Config implements Parcelable {
...
-           boolean isParentalGateEnabled,
-           boolean isBumperPageEnabled,
-           boolean shouldShowSmallClick,
-           boolean isBackButtonEnabled,
-           boolean shouldCloseAtEnd,
-           boolean shouldMuteOnStart,
-           CloseButtonState closeButtonState,
-           boolean shouldShowCloseWarning,
-           SAOrientation orientation) {
+    VideoConfig(boolean shouldShowPadlock,
```
Comment (gunhansancar) [inline superawesome-base/src/main/java/tv/superawesome/sdk/publisher/SAVideoActivity.java:336]: I really don't see why we rename this here?

File: app/src/main/java/tv/superawesome/demoapp/MyApplication.kt (hunk 1 above)
Comment (me2k) [reply app/src/main/java/tv/superawesome/demoapp/MyApplication.kt:16]: This was from fixing the AA Javascript issue. I'll revert it now. Interestingly, changing this doesn't work.

File: superawesome-base/src/main/java/tv/superawesome/sdk/publisher/SAVideoActivity.java (hunk 2 above)
Comment (me2k) [reply superawesome-base/src/main/java/tv/superawesome/sdk/publisher/SAVideoActivity.java:336]: It's to distinguish between `VideoConfig` and `ManagedAdConfig`

File: superawesome-base/src/main/java/tv/superawesome/sdk/publisher/managed/ManangedAdConfig.kt [hunk 3]
```diff
@@ -5,25 +5,30 @@ import android.os.Parcelable
...
     constructor(input: Parcel) {
+        isBumperPageEnabled = input.readByte().toInt() != 0
         isBackButtonEnabled = input.readByte().toInt() != 0
         closeButtonState = CloseButtonState.fromInt(input.readInt())
     }
 
     override fun describeContents(): Int = 0
 
     override fun writeToParcel(parcel: Parcel, i: Int) {
+        parcel.writeByte((if (isBumperPageEnabled) 1 else 0).toByte())
```
Comment (gunhansancar) [inline superawesome-base/src/main/java/tv/superawesome/sdk/publisher/managed/ManangedAdConfig.kt:31]: We can use `writeBoolean()`

Comment (me2k) [reply superawesome-base/src/main/java/tv/superawesome/sdk/publisher/managed/ManangedAdConfig.kt:31]: This requires API level 29

**Pull request merged**